gnucash/report/report-core.scm

```diff
@@ -748,7 +749,10 @@ not found.")))
 ;; where captured-error is the error string.
 (define (gnc:render-report report)
   (define (get-report) (gnc:report-render-html report #t))
-  (gnc:apply-with-error-handling get-report '()))
+  (call-with-time-limit
+   10
```

Comment:
I'm concerned with the hard-coded time limit. Generating a complex report on less capable hardware may easily exceed this.
Some way for the user to control what happens should be provided.
My ideal case would be what Firefox currently does when a javascript takes too long to respond: it will inform the user via a popup and offers the option to either continue to wait or to abort. This is ideal because the long-running process is not reset by default. Only on user request.

Reply:
Sure. In this case I do not think there's anything guile has to offer unless we allow multi threading in guile. This branch was mainly to illustrate a new guile capability. Additionally looking into guile sources, the sandbox uses sigalarm currently but they want to use thread, so, we shouldn't use it.